cmd/swarm/swarm-smoke/sliding_window.go

```diff
@@ -17,58 +17,74 @@
 package main
 
 import (
-	"crypto/md5"
-	crand "crypto/rand"
+	"bytes"
 	"fmt"
-	"io"
 	"math/rand"
 	"time"
 
 	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/metrics"
+	"github.com/ethereum/go-ethereum/swarm/testutil"
 	"github.com/pborman/uuid"
 
 	cli "gopkg.in/urfave/cli.v1"
 )
 
-var seed = time.Now().UTC().UnixNano()
-
-func init() {
-	rand.Seed(seed)
-}
-
 type uploadResult struct {
 	hash   string
 	digest []byte
 }
 
-func slidingWindow(c *cli.Context) error {
-	generateEndpoints(scheme, cluster, appName, from, to)
+func slidingWindow(ctx *cli.Context, tuid string) error {
+	errc := make(chan error)
+
+	go func() {
+		errc <- sl(ctx, tuid)
```

Comment:
I think short function names are allowed when scope is limited and near, which is the case here.

`sl` stands for `sliding window` and the function is in the near-vicinity of the main `slidingWindow` function.

Reply:
@holisticode http://doc.cat-v.org/bell_labs/pikestyle - this applies to Go as well.

Reply:
https://talks.golang.org/2014/names.slide#4

Reply:
I would agree with @holisticode. `sl` and similar functions here are in the package scope, it would be nice to have a descriptive name since every object in this package scope have a descriptive name.

Reply:
What would you suggest for this function, considering that `slidingWindow` is taken?

Reply:
I would suggest to call this function slidingWindow and the current one slidingWindowCmd, and all functions that have `func(*Context, string) error` signature to have Cmd suffix. The actual implementation of sliding window is in sl, while current slidingWindow function only handles timeout.

Reply:
Names are changed now.